core/src/datapoint_source/assets_exchange_rate.rs

```diff
@@ -32,3 +36,15 @@ pub struct AssetsExchangeRate<PER1: Asset, GET: Asset> {
     pub get: GET,
     pub rate: f64,
 }
+
+// Calculates an Exchange Rate of GET/PER2 based on GET/PER1 and PER1/PER2
+pub fn convert<GET: Asset, PER1: Asset, PER2: Asset>(
```

Comment:
This allows for type-safe conversions between exchange rates. Not sure if convert is the best name for this function however

Reply:
Nice! I think `convert` is too generic. How about `convert_rate`?